docs/notes-prove-verify.md

```diff
@@ -1,6 +1,392 @@
+can be provided by the user 
+depending on the circuit 
+composition. 
+For an addition gate, 
+we derive it as follows:
+
+\\[
+\begin{aligned}
+\mathbf{Q}\_*L\_i* = 1
+\mathbf{Q}\_*R\_i* = 1
+\mathbf{Q}\_*0\_i* = -1
+\mathbf{Q}\_*M\_i* = 0
+\mathbf{Q}\_*C\_i* = 1
+\end{aligned}
+\\]
+
+Which results in:
+\\[
+\begin{aligned}  
+a\_i +
+b\_i -
+c\_i =
+0
+\end{aligned}
+\\]
+
+For a multiplication gate, 
+we derive it as follows:
+
+\\[
+\begin{aligned}
+\mathbf{Q}\_*L\_i* = 0
+\mathbf{Q}\_*R\_i* = 0
+\mathbf{Q}\_*0\_i* = -1
+\mathbf{Q}\_*M\_i* = 1
+\mathbf{Q}\_*C\_i* = 0
+\end{aligned}
+\\]
+
+Which results in:
+\\[
+\begin{aligned}  
+a\_ib\_i -
+c\_i =
+0
+\end{aligned}
+\\]
+
+With this format, there is 
+a specific method used to 
+convert all the equations 
+into polynomials form.
+Basically, for bundling
+these together, PLONK can 
+take sets of equations and 
+turn them into one single 
+equation over polynomials. 
+This called the evalutaion 
+form. We are then able to 
+use Lagrangian interpolation
+to convert coefificient form. 
+All this interpolation is doing
+is allowing us to evaluate a
+functions over specific points,
+for 'x' values,the polynomial 
+is equal to '1' or '0'.
+
+With these specific basis, we
+can derive the relation between 
+all sets fo equations into one 
+single polynomial equation,
+where we have a vector of inputs
+to each gate type:
+\\[
+\begin{aligned}
+\mathbf{Q}\_*L*(*x*) \circ a(*x*) +
+\mathbf{Q}\_*R*(*x*) \circ b(*x*) +
+\mathbf{Q}\_*0*(*x*) \circ c(*x*) +
+\mathbf{Q}\_*M*(*x*) \circ a(*x*)b(*x*) +
+\mathbf{Q}\_*C*(*x*) =
+0
+\end{aligned}
+\\]
+
+The utility for this is PLONK, 
+as a univeral SNARK, is that 
+any operaation or relationship 
+that holds with the vectors, 
+will also hold over the polynomial.
+
+In order to check this in PLONK, 
+a 'vanishing polynomial' is 
+introduced. Which is just a 
+polynomial equal to zero for
+all the points we evaluate at.
+So this means that the vectors
+will be divisible by this vanashing
+polynomial, if the expression we 
+check does indeed hold. 
+
+To summarise what the PLONK
+proof must satisfy:
+
+The generation of copy and 
+gate constraints where the 
+former are representative 
+of permuted wires. Generate 
+all of the polynomials to  
+be checked against the 
+vanishing polynomials.
+Take all of the wire values
+and convert them into three
+polynomials, A(x), B(x), C(x). 
+Check the polynomials at 
+some random Z(*z*), by making
+commitments and checking 
+the evaluation form. 
+Then commit all evalution 
+polynomials forms for the 
+verfier. 
+
+
+
+
+
+
+
+
+
+
+The use of polynomials in the 
+PLONK proving scheme refers
+to specific evaluation domains,
+names Lagrangian polynomials,  
+based on interpolation of two 
+functions of a particular group
+elements. The following section 
+gives a more comprehensive
+understanding to the way in 
+which these polynomials are 
+formed, given certain inputs. 
+
+
+Langrangian polynomials are 
```

Comment:
Shouldn't we add a tittle or something similar?

Reply:
Well, this is the separate part, where we add in possibly else where. Even still, you're right, i'll add one :)